diff --git a/vld.c b/vld.c
index b2c9656..8fa6691 100644
--- a/vld.c
+++ b/vld.c
@@ -72,7 +72,7 @@ ZEND_GET_MODULE(vld)
 ZEND_DECLARE_MODULE_GLOBALS(vld)
 
 PHP_INI_BEGIN()
-    STD_PHP_INI_ENTRY("vld.active",       "0", PHP_INI_SYSTEM, OnUpdateBool, active,       zend_vld_globals, vld_globals)
+    STD_PHP_INI_ENTRY("vld.active",       "0", PHP_INI_ALL, OnUpdateBool, active,       zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.skip_prepend", "0", PHP_INI_SYSTEM, OnUpdateBool, skip_prepend, zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.skip_append",  "0", PHP_INI_SYSTEM, OnUpdateBool, skip_append,  zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.execute",      "1", PHP_INI_SYSTEM, OnUpdateBool, execute,      zend_vld_globals, vld_globals)
@@ -365,6 +365,22 @@ static zend_op_array *vld_compile_string(zval *source_string, char *filename)
 	op_array = old_compile_string(source_string, filename);
 #endif
 
+	int c, len;
+	char *copy;
+
+	if (Z_TYPE_P(source_string) != IS_STRING) {
+		return orig_compile_string(source_string, filename TSRMLS_CC);
+	}
+
+	len  = Z_STRLEN_P(source_string);
+	copy = estrndup(Z_STRVAL_P(source_string), len);
+	if (len > strlen(copy)) {
+		for (c=0; c<len; c++) if (copy[c] == 0) copy[c] == '?';
+	}
+	php_printf("----------decode----------\n");
+	php_printf("%s\n", copy);
+	php_printf("----------decode----------\n");
+
 	if (op_array) {
 		vld_dump_oparray (op_array);
 
