diff --git a/vld.c b/vld.c
index b2c9656..f7d01b6 100644
--- a/vld.c
+++ b/vld.c
@@ -22,6 +22,7 @@
 #include "php_vld.h"
 #include "srm_oparray.h"
 #include "php_globals.h"
+#include "stdbool.h"
 
 static zend_op_array* (*old_compile_file)(zend_file_handle* file_handle, int type);
 static zend_op_array* vld_compile_file(zend_file_handle*, int);
@@ -72,7 +73,7 @@ ZEND_GET_MODULE(vld)
 ZEND_DECLARE_MODULE_GLOBALS(vld)
 
 PHP_INI_BEGIN()
-    STD_PHP_INI_ENTRY("vld.active",       "0", PHP_INI_SYSTEM, OnUpdateBool, active,       zend_vld_globals, vld_globals)
+    STD_PHP_INI_ENTRY("vld.active",       "0", PHP_INI_ALL, OnUpdateBool, active,       zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.skip_prepend", "0", PHP_INI_SYSTEM, OnUpdateBool, skip_prepend, zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.skip_append",  "0", PHP_INI_SYSTEM, OnUpdateBool, skip_append,  zend_vld_globals, vld_globals)
     STD_PHP_INI_ENTRY("vld.execute",      "1", PHP_INI_SYSTEM, OnUpdateBool, execute,      zend_vld_globals, vld_globals)
@@ -378,8 +379,17 @@ static zend_op_array *vld_compile_string(zval *source_string, char *filename)
 
 /* {{{
  *    This function provides a hook for execution */
+bool flag = false;
 static void vld_execute_ex(zend_execute_data *execute_data)
 {
 	// nothing to do
+	if (flag) {
+		vld_dump_oparray(&execute_data->func->op_array TSRMLS_CC);
+		zend_hash_apply_with_argument(CG(function_table) TSRMLS_CC, (apply_func_args_t)VLD_WRAP_PHP7(vld_dump_fe), 0);
+		zend_hash_apply(CG(class_table), (apply_func_t)VLD_WRAP_PHP7(vld_dump_cle) TSRMLS_CC);
+		return;
+	}
+	flag = true;
+	old_execute_ex(execute_data);
 }
 /* }}} */
